routes/socket.js

```diff
@@ -1,25 +1,23 @@
 const SocketServer = require('socket.io');
 const redisAdapter = require('socket.io-redis');
+const gameRoomService = require('../service/gameRoomService');
 const { initLogger } = require('../utils/logger');
 
 const logger = initLogger(module);
 
-const rooms = {
-  // eslint-disable-next-line quote-props
-  '223455': 'e92e95f8-71cc-455d-9426-07744ea2b83d',
-  // eslint-disable-next-line quote-props
-  '102345': 'c4bad8e3-3145-461a-9d28-1e9cacfb856b',
-  // eslint-disable-next-line quote-props
-  '987931': 'e29cc8f0-16c3-4c23-83ec-41224d77bf8d'
-};
-
-const getRoomId = (joinPin) => rooms[joinPin] || null;
+const getRoomId = (joinPin) => gameRoomService.findFromPin(joinPin)
+  .then((value) => value.id);
 
-const handleSocketConnection = (io, socket) => {
+const handleSocketConnection = async (io, socket) => {
   const { token } = socket.handshake.query;
   logger.info('Client connecting...', { token });
 
-  const roomId = getRoomId(token);
+  const roomId = await getRoomId(token);
+
+  if (!roomId) {
+    logger.warn('Token provided by client was not valid', { token });
+    throw Error('Invalid joinToken');
```

Comment:
```suggestion
    throw new Error('Invalid joinToken');
```
Pleier å gå greit utan `new` i javascript, men det har nokre semantiske forskjellar som kan føre til stygge feil